### PR TITLE
added licenses and fixed py ver

### PR DESCRIPTION
**Pull request by buijennifer**
## Description

This PR was initially opened to add a missing license as well as correct the `skip` field in the recipe. The package was built for python versions _less than_ 3.6 vs _greater than_ 3.6. 
After fixing these mistakes, I found that the feedstock is missing the `conda-forge.yml` file. Thus, I have added that. 
Additionally, this PR attempts to build the package as noarch. 

## Motivation and Context

This package is a dependency of `contextvars`, which is a dependency for `caproto`. The ultimate goal is to build `ophyd` which depends on `caproto`. 

## How Has This Been Tested?

The changes will be tested via Azure.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,9 +13,9 @@ source:
   '{{ hash_type }}': '{{ hash_value }}'
 
 build:
-  number: 0
+  number: 1
   script: "{{ PYTHON }} -m pip install . --no-deps -vv"
-  skip: True # [py>36]
+  skip: True # [py<36]
```
Comment (mrakitin) [inline recipe/meta.yaml:18]: Not sure why it's showing this diff, the master branch has a correct condition now: https://github.com/nsls-ii-forge/immutables-feedstock/blob/b91e50415117e1e9146f93f2f98e5e0314ee621c/recipe/meta.yaml#L18

Any ideas?

**Review by mrakitin — COMMENTED** [review]
Can it be made `noarch` as well?

File: recipe/meta.yaml (hunk 1 above)
Comment (buijennifer) [reply recipe/meta.yaml:18]: When I initially rendered with the wrong sign. I think some of my forced pushes made the history confusing.

**Comment by buijennifer** [comment]
Is it okay to just add noarch to the recipe? The conda-forge recipe isn't noarch and there are no noarch files for this package on anaconda.org.

File: recipe/meta.yaml (hunk 1 above)
Comment (mrakitin) [reply recipe/meta.yaml:18]: OK, thanks.

**Comment by mrakitin** [comment]
Could you please resolve the conflicts (=rebase on master)?

**Comment by mrakitin** [comment]
> Is it okay to just add noarch to the recipe? The conda-forge recipe isn't noarch and there are no noarch files for this package on anaconda.org.

Sorry, I didn't check the source code of the project. It has C++ extensions (see https://github.com/MagicStack/immutables/blob/1a51a34154654285e017e6006d16c03cbee92637/setup.py#L62), so it has to be arch-dependent.

Also, I noticed that the temp vim file was added to the version control: https://github.com/nsls-ii-forge/immutables-feedstock/blob/master/recipe/.meta.yaml.swo. Pease remove that file.

**Comment by buijennifer** [comment]
What is the best way to resolve the conflict? Should I change the recipe on the master branch?

**Comment by mrakitin** [comment]
> What is the best way to resolve the conflict? Should I change the recipe on the master branch?

```
git checkout master
git pull --all
git checkout <your PR branch>
git rebase master
# fix the conflicts by editing the files with changes in both master and the PR branch
git add <the conflicting file>
git rebase --continue
git push --force-with-lease
```

**Comment by buijennifer** [comment]
What I did replace the `skip` field with `noarch` in the master recipe. I hope that is okay too.

**Comment by mrakitin** [comment]
Changing the master branch directly is not recommended as it's considered the "ground truth". It should be done on the PR level.

**Comment by buijennifer** [comment]
Now I am confused. What I did was replace the `noarch` field with the `skip` field, but Azure is building a noarch recipe?

**Comment by buijennifer** [comment]
I am confused. My last commit was replacing `noarch` with `skip` field. The recipe now has `skip` field, but why is Azure building a noarch recipe?

**Comment by mrakitin** [comment]
Did you rerender again after the changes?

**Comment by buijennifer** [comment]
Right, I forgot to rerender. My apologies, again.

**Comment by mrakitin** [comment]
It would be nice to have some kind of checklist to do for each change (like bumping a build number, checking for possibility to convert to `noarch`, rerendering, etc.) It can be done with the templates for the PRs and issues, see an example at https://github.com/bluesky/bluesky/tree/master/.github. Do you mind implementing it?

**Review by mrakitin — APPROVED** [review]

**Pull request merged**